### PR TITLE
added code docu

### LINKED ISSUE #30: Code Documentation

### PR DESCRIPTION
**Pull request by Hannes01071995**
Wrote doxygen docu for our interface.

closes #30

### CODE REVIEW

File: src/node_lib.h [hunk 1]
```diff
@@ -104,37 +159,74 @@ namespace node { namespace lib {
...
 
-    /*
-    Adds a new JavaScript module to the *running* Node.js engine.
-    */
+    /**
+     * @brief Adds a NPM module to the current JavaScript context.
+     * 
+     * Adds a given NPM module to the JavaScript context.
+     * This is achieved by calling `require('module_name')`.
+     * *Important* Make sure the NPM module is installed before using this method.
```
Comment (emberflare) [inline src/node_lib.h]: Where does this module need to be installed? Globally? Or locally (which folder though?)?

File: src/node_lib.h [hunk 2]
```diff
@@ -9,93 +9,148 @@
...
-    /*
-    Executes a given JavaScript file and returns once the execution has finished.
-    *Important*: Node.js has to have been initialized by calling Initialize().
-    */
+    /**
+     * @brief Executes the content of a given JavaScript file.
+     * 
+     * Loads and executes the content of the given file.
+     * This method returns after the script was evaluated once.
+     * This means, that any pending events will not be processes as long as
```
Comment (emberflare) [inline src/node_lib.h]: processed*

File: src/node_lib.h [hunk 3]
```diff
@@ -9,93 +9,148 @@
...
 
-    /*
-    Stops the Node.js event loop after its current execution. Execution can be resumed by calling RunEventLoop() again.
-    */
+    /**
+     * @brief Issues the Node.js event loop to stop.
+     * 
+     * Issues the Node.js event loop to stop.
+     * The event loop will finish its current execution.
+     * This means, that the loop is not stopped when this method returns.
```
Comment (emberflare) [inline src/node_lib.h]: Suggestion for more clarity: "...that the loop is not guaranteed to have stopped when this method returns."

File: src/node_lib.h (hunk 1 above)
Comment (cmfcmf) [reply src/node_lib.h]: Here's how node finds its modules: https://nodejs.org/api/modules.html#modules_all_together
Note: We haven't properly solved this with our qt app (yet).

**Comment by emberflare** [comment]
closes #30

**Pull request merged**